docs(planning_design): update for internal interface (#371)
autowarefoundation/autoware-documentation: docs/design/autoware-architecture/planning/index.md

- **To API Layer**
  - Planning factors: Provides information about the reasoning behind the current planning behavior. This may include the position of target objects to avoid, obstacles that led to the decision to stop, and other relevant information.

### Internal interface in the planning component

- **Mission Planning to Scenario Planning**
  - Route: Offers guidance for the path that needs to be followed from the starting point to the destination. This path is determined based on information such as lane IDs defined on the map. At the route level, it doesn't explicitly indicate which specific lanes to take, and the route can contain multiple lanes.
- **Behavior Planning to Motion Planning**

>   - Path: Provides a rough position and velocity to be followed by the vehicle. These path points are defined with an approximate interval of about 1 meter, although it's not strictly limited to that.

Reviewer: The part after ", although ..." sounds a bit weird. How about adding a bit more details like the following:
```suggestion
  - Path: Provides a rough position and velocity to be followed by the vehicle. These path points are usually defined with an interval of about 1 meter. Although other interval distances are possible, it may impact the precision or performance of the planning component.
```

  - Drivable area: Defines regions where the vehicle can drive, such as within lanes or physically drivable areas. It assumes that the motion planner will calculate the final trajectory within this defined area.
- **Scenario Planning to Validation**

>   - Trajectory: Defines the desired position, velocity, and acceleration which the Control Component calculates the control command to follow. Trajectory points are defined at intervals of approximately 0.1 seconds, considering the vehicle's speed.

Reviewer: ```suggestion
  - Trajectory: Defines the desired positions, velocities, and accelerations which the Control Component will try to follow. Trajectory points are defined at intervals of approximately 0.1 seconds based on the trajectory velocities.
```

- **Validation to Control Component**

>   - Trajectory: Same as above, with safety consideration.

Reviewer: ```suggestion
  - Trajectory: Same as above but with some additional safety considerations.
```
